### PR TITLE
test: add consumer-producer test for condvar

### PR DESCRIPTION
**Pull request by shigeki**
Conditional variable is newly supported in libuv so it is nice to have a consumer-producer test with 32 consumer threads to check racing. 
The test is really working fine on Windows, of cource on Linux and SmartOS. Thanks for the good job.

### CODE REVIEW

File: test/test-condvar-consumer-producer.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  int data;
+};
+typedef struct buffer_s buffer_t;
+
+static ngx_queue_t queue;
+static uv_mutex_t mutex;
+static uv_cond_t empty;
+static uv_cond_t full;
+
+volatile int finished_consumers = 0;
```
Comment (bnoordhuis) [inline test/test-condvar-consumer-producer.c:42]: Should be static.

File: test/test-condvar-consumer-producer.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+
+void produce(int value) {
+  buffer_t* buf;
+  buf = malloc(sizeof(buffer_t));
```
Comment (bnoordhuis) [inline test/test-condvar-consumer-producer.c:47]: sizeof(*buf)

File: test/test-condvar-consumer-producer.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  ngx_queue_init(&buf->queue);
+  (*buf).data = value;
```
Comment (bnoordhuis) [inline test/test-condvar-consumer-producer.c:49]: buf->data?

File: test/test-condvar-consumer-producer.c [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  ASSERT(ngx_queue_empty(&queue));
```
Comment (bnoordhuis) [inline test/test-condvar-consumer-producer.c:50]: Doesn't hurt but kind of superfluous. :-)

File: test/test-condvar-consumer-producer.c [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  ngx_queue_insert_tail(&queue, &buf->queue);
+}
+
+
+int consume() {
```
Comment (bnoordhuis) [inline test/test-condvar-consumer-producer.c:55]: static void consume(void) {

File: test/test-condvar-consumer-producer.c [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  ngx_queue_t *q;
+  buffer_t *buf;
```
Comment (bnoordhuis) [inline test/test-condvar-consumer-producer.c:57]: Style: type\* name

**Comment by bnoordhuis** [comment]
Never mind all the comments, I landed your patch with style fixups in f7c991f. :-)

By the way, I made it free the buffer so valgrind stops complaining. Thanks, Shigeki.

**Comment by shigeki** [comment]
@bnoordhuis What a memory leak I made! You are my best teacher. Thanks.

**Pull request closed without merging**